public/electron.js

```diff
@@ -33,8 +34,65 @@ function createWindow() {
     mainWindow.show();
   });
   ipcHandler(mainWindow);
+
+  mainWindow.on("minimize", (e) => {
+    e.preventDefault();
+    mainWindow.hide();
+  });
+
+  mainWindow.on("close", (e) => {
+    if (!app.isQuiting) {
+      e.preventDefault();
+      mainWindow.hide();
+      tray.setContextMenu(trayMenuWithShow);
+    }
+  });
 }
 
+const trayMenuWithShow = Menu.buildFromTemplate([
+  {
+    label: "Show",
+    click: () => {
+      mainWindow.show();
+      tray.setContextMenu(trayMenuWithHide);
+    },
+  },
+  {
+    label: "Quit",
+    click: () => {
+      app.isQuiting = true;
+      app.quit();
+    },
+  },
+]);
+
+const trayMenuWithHide = Menu.buildFromTemplate([
+  {
+    label: "Hide",
+    click: () => {
+      mainWindow.hide();
+      tray.setContextMenu(trayMenuWithShow);
+    },
+  },
+  {
+    label: "Quit",
+    click: () => {
+      app.isQuiting = true;
+      app.quit();
+    },
+  },
```

Comment:
A minor suggestion: perhaps we could avoid duplicating the `Quit` option - it's currently defined in both menus.

Reply:
I wanted to make it dynamic at first, but as far as I researched tray menu can't be dynamic, so the solution is to create two separate context menus (one with Hide and other with Show option).

Reply:
Try moving the whole quit menu item object to a variable that can be used in both menus. Should work like that.

```
const quitMenuItem = {
    label: "Quit",
    click: () => {
      app.isQuiting = true;
      app.quit();
    };
};
```